lib/ProductOpener/Ingredients.pm

```diff
@@ -963,87 +1016,115 @@ sub parse_specific_ingredients_from_text ($product_ref, $text, $percent_or_quant
+					)
 				)
-			{
-				$ingredient = $1;
-				# 2 groups captured by $percent_or_quantity_regexp:
-				$percent_or_quantity_value = $2;
-				$percent_or_quantity_unit = $3;
-				$matched_text = $&;
-				# Remove the matched text
-				$text = $` . ' ' . $';
-			}
 
-			# Origin of the milk: United Kingdom
-			elsif (match_origin_of_the_ingredient_origin($ingredients_lc, \$text, $matched_ingredient_ref)) {
-				$origins = $matched_ingredient_ref->{origins};
-				$ingredient = $matched_ingredient_ref->{ingredient};
-				$matched_text = $matched_ingredient_ref->{matched_text};
-				# Remove extra spaces
-				$ingredient =~ s/\s+$//;
-			}
-		}
-		elsif ($ingredients_lc eq "fr") {
+			)
+			)
+		{
+			$log->debug("parse_specific_ingredients_from_text - text in this order: ingredient - quantity - percent")
+				if $log->is_debug();
 
-			# examples:
-			# Teneur en lait 25% minimum.
-			# Teneur en lactose < 0,01 g/100 g.
-			# Préparée avec 50 g de fruits pour 100 g de produit fini.
+			my $before = $1;
+			$ingredient = $2;
+			# 2 groups captured by $percent_or_quantity_regexp:
+			$percent_or_quantity_value = $3;
+			$percent_or_quantity_unit = $4;
+			$matched_text = $&;
+			# Remove the matched text
+			$text = $` . $1 . ' ' . $';
 
-			if (
-				(defined $percent_or_quantity_regexp)
-				and ($text
-					=~ /\s*(?:(?:préparé|prepare)(?:e|s|es)? avec)(?: au moins)?(?::| )+$percent_or_quantity_regexp (?:de |d')?([^,.;]+?)\s*(?:$per_100g_regexp(?:[^,.;-]*?))?(?:;|\.| - |$)/i
+			$log->debug("parse_specific_ingredients_from_text - ingredient: $ingredient") if $log->is_debug();
+			$log->debug("parse_specific_ingredients_from_text - percent_or_quantity_value: $percent_or_quantity_value")
+				if $log->is_debug();
+			$log->debug("parse_specific_ingredients_from_text - percent_or_quantity_unit: $percent_or_quantity_unit")
+				if $log->is_debug();
+		}
+		# text in this order: quantity - ingredient - percent
+		# e.g.
+		# - 75g of tomatoes per 100g
+		# - prepared with 60g of fruits
+		# - prepared with 40g of fruits per 100g of finished product
+		elsif (
+			(defined $percent_or_quantity_regexp)
+			and (
+				# if the string does not start with "prepared with", it needs to finish with "per 100g",
+				# otherwise we will match items that could be part of the ingredients list such as "75% of tomatoes
+
+				# prepared with, percent, ingredient, optional per 100g, separator
+				# $of needs to be first in (?:$of|\s|:) so that " of " is matched by it, instead of the ingredient capturing group
+				(
+					$text
+					=~ /((?:^|;|,|\.| - )\s*)$prepared_with(?:$of|\s|:)+$percent_or_quantity_regexp(?:$of|\s|:)+\s*([^,.;]+?)\s*(?:$per_100g_regexp)?(?:;|,|\.| - |$)/i
 				)
+				or
+				# percent, ingredient, per 100g, separator
+				(
+					$text
+					=~ /((?:^|;|,|\.| - )\s*)$percent_or_quantity_regexp(?:$of|\s|:)+\s*([^,.;]+?)\s*(?:$per_100g_regexp)(?:;|,|\.| - |$)/i
```

Comment:
```suggestion
					=~ /((?:^|;|,|\.| - )\s*)$percent_or_quantity_regexp(?:$of|\s|:)+\s*([^,.;]+?)\s*(?:$per_100g_regexp)?(?:;|,|\.| - |$)/i
```

Reply:
@benbenben2 in this case, the "per 100g" is not optional (see comment above: # if the string does not start with "prepared with", it needs to finish with "per 100g", )
That's because otherwise, we will match things like "some ingredient 56%" which is just part of the normal ingredient list.